### ISSUE TITLE
Command-line interface for whatshap learn

### ISSUE THREAD

**Issue by marcelm** [whatshap/whatshap#490]
Hi @Hufsah-Ashraf and @tobiasmarschall. I’m thinking that it would be time for a new release considering #489 reports a bug that we already fixed.

Would it be ok to include kmerald and `whatshap learn` in a 2.1 release? If not, I can open a branch for 2.1 that does not contain the changes from #481.

In any case, I’m reviewing the `whatshap learn` command, which was added in #481. I noticed it’s a bit cumbersome to use because it needs a lot of parameters, all of which need to be typed out (because of `required=True`), and it’s unclear how to select some of them.

* [x] Are there any sensible defaults for *k* and the window size so that they do not have to be `required=True`?
* [x] If not, the documentation needs to explain how to choose these values.
* [x] The other WhatsHap commands accept the BAM and VCF files as positional arguments (without `--bam`/`--vcf`)
* [x] `--kmer` should be named `-k` (it’s used to specify *k*, not a *k*-mer)
* [x] `--output` should have a short version `-o` for consistency with the other WhatsHap commands
* [x] `--reference` should have a short version `-r` for the same reason
* [x] The description for `--window` makes me think that the actual window size (as I would understand it) is actually two times `--window` (perhaps plus one). Which one is it?

I’m going to make some of these changes myself, but in particular would appreciate advice on good defaults for *k* and the window size.

**Comment by tobiasmarschall** [comment]
Thanks, @marcelm! Good points. Would be nice to address them and include k-merald in the next release. @Hufsah-Ashraf Could you address these points? I'd vote to set a default for `k`, e.g. 5 or 7.

**Comment by marcelm** [comment]
Thanks! I’ve checked the points above that are taken care of by PR #491.

**Comment by Hufsah-Ashraf** [comment]
Thanks @marcelm. I agree with @tobiasmarschall that it would be nice to include k-merald in the next release. 
I've now addressed the rest of your points. For k and window size, I have set the default values to those I've been using for the experiments we did for the paper i.e. 7 and 25, respectively. Re the window size, you understood correctly i.e. the actual window size would be twice the specified value. I updated the description a little bit, I hope it would be clearer now. I've pushed the changes to 'learn-refactor' branch. However, I mistakenly pushed them to the 'main' first. Apologies for that and I hope it doesn't cause too much inconvenience.
